spec/withTimeoutSpec.js

```diff
@@ -0,0 +1,42 @@
+const withTimeout = require('../src/statics/withTimeout');
+const TimeoutError = require('../src/statics/TimeoutError');
+
+
+describe('withTimeout', () => {
```

Comment:
It might make sense to use `lolex` or `timekeeper` for these tests (and maybe `jest`) for isolation - these tests _should_ behave appropriately but some engines have pretty weak timing and ordering guarantees

Reply:
Since this test suite is in Jasmine and the change is already an add-on this is approaching my scope creep limit. I will increase the margin to prevent out of order timeouts.